interop-tests/src/test/java/example/myapp/helloworld/grpc/RichErrorModelTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2018-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package example.myapp.helloworld.grpc;
+
+import akka.actor.ActorSystem;
+import akka.grpc.GrpcClientSettings;
+import akka.grpc.Trailers;
+import akka.grpc.internal.GrpcMetadataImpl;
+import akka.grpc.javadsl.GrpcExceptionHandler;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.ServerBinding;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.model.HttpResponse;
+import akka.japi.Function;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.protobuf.StatusProto;
+import org.junit.Assert;
+import org.junit.Test;
+import org.scalatestplus.junit.JUnitSuite;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class RichErrorModelTest extends JUnitSuite {
+
+    private com.google.protobuf.any.Any fromJavaProto(com.google.protobuf.Any javaPbSource) {
+        return com.google.protobuf.any.Any.of(javaPbSource.getTypeUrl(), javaPbSource.getValue());
+    }
+
+    private CompletionStage<ServerBinding> run(ActorSystem sys) throws Exception {
+
+        GreeterService impl = new RichErrorImpl();
+
+        // #custom_eHandler
+        Function<ActorSystem, Function<Throwable, Trailers>> customErrorHandler = new Function<ActorSystem, Function<Throwable, Trailers>>() {
+            @Override
+            public Function<Throwable, Trailers> apply(ActorSystem param) throws Exception, Exception {
+                return new Function<Throwable, Trailers>() {
+                    @Override
+                    public Trailers apply(Throwable ex) throws Exception, Exception {
+                        if ((ex instanceof CompletionException)) ex = ex.getCause();
+                        if ((ex instanceof io.grpc.StatusRuntimeException)) {
+                            StatusRuntimeException statusEx = (StatusRuntimeException) ex;
+                            return new Trailers(statusEx.getStatus(), new GrpcMetadataImpl(statusEx.getTrailers()));
```

Comment:
hmm, `GrpcMetadataImpl` is from `akka.grpc.internal` - don't we have a user-facing way to get an instance yet?

Reply:
I did not find anything into that direction.

Reply:
I think we should change the default exception mapper to do this. I don't mind changing that in a different PR, but once that's changed the documentation for the rich error model can be simplified a bit

Reply:
Great suggestion - the examples now do look much more concise